test/file.jl

```diff
@@ -47,9 +64,12 @@ end
 
 # test links
 @unix_only @test islink(link) == true
+@unix_only @test readlink(link) == file
+
 @non_windowsxp_only begin
     @test islink(dirlink) == true
     @test isdir(dirlink) == true
+    @unix_only @test readlink(dirlink) == subdir
```

Comment:
yeah pushing to this branch is good - if it gets to be too many commits we may ask you to squash later, but for now this is fine